src/Moq/Language/Flow/SetupSequencePhrase.cs

```diff
@@ -71,12 +71,34 @@ public ISetupSequentialResult<TResult> Returns(Func<TResult> valueFunction)
 			if (valueFunction is TResult)
 			{
 				return this.Returns((TResult)(object)valueFunction);
-			}
+			}	
 
 			this.setup.AddReturns(() => valueFunction());
 			return this;
 		}
 
+		public ISetupSequentialResult<TResult> ReturnsAsync(Func<TResult> valueFunction)
```

Comment:
This doesn't seem right for two reasons:

* You are defining this method on a type that is defined `internal`, without having any `public` counterpart. I would assume that it's actually not possible to call this added method. (Have you tried?)

* By putting this method here, you cannot distinguish between a setup on a method returning `Task`, `ValueTask`, or something else. If you want to make that distinction, you need to implement things as an extension method. See e.g. [the `ThrowsAsync` extension methods on `ISetupSequentialResult<[Value]Task<TResult>>` in `AsyncExtensions.cs`](https://github.com/moq/moq4/blob/bf19ad64540b9e3e71b8a869b2b1ab204ff1dbcb/src/Moq/AsyncExtensions.cs#L275-L299). I believe this is where this new method should be implemented.